Copy Colvars CUDA source code to all supported platforms

HanatoK opened the pull request:
To ensure that (i) NAMD can create the same Make.depends files across CPU and GPU builds, and (ii) the source code of Colvars to be MD engine-agnostic, I end up with copying the CUDA source code to all supported platforms, even if the code are not actually compiled (guarded by `!defined(COLVARS_CUDA) && !defined(COLVARS_HIP)`.

This PR depends on https://github.com/Colvars/colvars/pull/888 and should wait for https://github.com/Colvars/colvars/pull/888 to be merged.

HanatoK committed:
build!: reorder the header inclusion order

This is to ensure that NAMD could generate stable Make.depends.

HanatoK committed:
build: explicitly add ${COLVARS_SOURCE_DIR}/src as include dir

HanatoK committed:
build: fix the GROMACS build

HanatoK committed:
build: copy the Colvars CUDA files to VMD

HanatoK committed:
build: update src/Makefile to include the current path

HanatoK committed:
build: reorder the macro after the header file inclusion

HanatoK committed:
build: update NAMD build dependency files

HanatoK committed:
build: copy the Colvars CUDA source files to LAMMPS

jhenin commented on namd/lepton/Make.depends:
```diff
@@ -1,80 +1,153 @@
+	lepton/include/lepton/ExpressionTreeNode.h \
+	lepton/include/lepton/Operation.h \
+	lepton/include/lepton/ParsedExpression.h \
+	lepton/include/lepton/windowsIncludes.h \
```
Nice to see dependencies in alphabetical order!

HubLot reviewed (approved):
LGTM

HanatoK committed:
build: fix the Make.depends files for lepton and CUDAGM plugin

HanatoK committed:
build: fix the typo in Make.depends

HanatoK commented:
I still need to update the NAMD build files to ensure that it doesn't break the CPU build.

HanatoK committed:
build: update NAMD build files

This commit moves all object files of the CUDAGM plugin into its own
Makefile.namd, and conditionally compiles them when NAMD is built with
NVCC.
